src/rank_llm/rerank/listwise/swerank_reranker.py

```diff
@@ -0,0 +1,391 @@
+            return self._vllm_handler.chat_completions(
+                prompts=prompts, max_tokens=max_tokens, temperature=0
+            )
+
+        outputs = self._vllm_handler.generate_output(
+            prompts=prompts,
+            min_tokens=max_tokens,
+            max_tokens=max_tokens,
+            temperature=0.0,
+        )
+        return [
+            (output.outputs[0].text, len(output.outputs[0].token_ids))
+            for output in outputs
+        ]
+
+    def run_llm(
+        self,
+        prompt: Union[str, List[Dict[str, str]]],
+        current_window_size: Optional[int] = None,
+    ) -> Tuple[str, int]:
+        """
+        Run inference on a single prompt.
+        """
+        if current_window_size is None:
+            current_window_size = self._window_size
+
+        max_tokens = self.num_output_tokens(current_window_size)
+
+        if self._base_url:
+            results = self._vllm_handler.chat_completions(
+                prompts=[prompt], max_tokens=max_tokens, temperature=0
+            )
+            return results[0]
+
+        output = self._vllm_handler.generate_output(
+            prompts=[prompt],
+            min_tokens=max_tokens,
+            max_tokens=max_tokens,
+            temperature=0.0,
+        )[0]
+        return output.outputs[0].text, len(output.outputs[0].token_ids)
+
+    def create_prompt_batched(
+        self,
+        results: List[Result],
+        rank_start: int,
+        rank_end: int,
+    ) -> List[Tuple[Union[str, List[Dict[str, str]]], int]]:
+        """
+        Create prompts for a batch of results.
+        """
+
+        def chunks(lst, n):
+            """Yield successive n-sized chunks from lst."""
+            for i in range(0, len(lst), n):
+                yield lst[i : i + n]
+
+        all_completed_prompts = []
+
+        with ThreadPoolExecutor() as executor:
+            for batch in chunks(results, self._batch_size):
+                completed_prompts = list(
+                    executor.map(
+                        lambda result: self.create_prompt(result, rank_start, rank_end),
+                        batch,
+                    )
+                )
+                all_completed_prompts.extend(completed_prompts)
+        return all_completed_prompts
+
+    def create_prompt(
+        self, result: Result, rank_start: int, rank_end: int
+    ) -> Tuple[Union[str, List[Dict[str, str]]], int]:
+        """
+        Create a prompt for ranking using the inference handler.
+        
+        This method uses different document length limits based on rerank type:
+        - Code: 1024 tokens (longer for code functions)
+        - Text: 300 tokens (standard for passages)
+        """
+        query = result.query.text
+        num = len(result.candidates[rank_start:rank_end])
+
+        # Set max document length based on rerank type
+        if self._rerank_type == "code":
+            max_doc_length = 1024  # Longer for code
```

Comment:
Where does this numbers come from?

We normally start with the user provided context size / window size